### PR TITLE
added global state and refactored to accomodate the state

### PR DESCRIPTION
**Pull request by a-miscellaneous**

### CODE REVIEW

**Comment by akriese** [comment]
@a-miscellaneous could you slide in a small fix into this? When executing the GUI, the preset paths are wrong. They should be test/... instead of main/... :)

**Comment by akriese** [comment]
Am I right, that the `state.value.copy(args)` args are only what changes in the state change, so that one does not have to write too much code? I yes, this approach seems good enough to me :)

**Review by AlperK61 — APPROVED** [review]
it still woks

File: GUI/src/main/kotlin/frameNavigation/FrameNavigation.kt [hunk 1]
```diff
@@ -18,14 +18,14 @@ import kotlin.math.min
  * @param paths [AllVideos] object containing the paths to the videos.
```
Comment (simonsasse) [inline GUI/src/main/kotlin/frameNavigation/FrameNavigation.kt:18]: adjust doc

File: GUI/src/main/kotlin/models/AppState.kt [hunk 2]
```diff
@@ -0,0 +1,23 @@
+package models
+
+import Screen
+import algorithms.AlignmentElement
+import java.nio.file.FileSystems
+
```
Comment (simonsasse) [inline GUI/src/main/kotlin/models/AppState.kt:6]: create doc

File: GUI/src/main/kotlin/ui/screens/DiffScreen.kt [hunk 3]
```diff
@@ -18,7 +18,7 @@ import androidx.compose.ui.input.key.*
 import androidx.compose.ui.res.painterResource
 import androidx.compose.ui.unit.dp
 import frameNavigation.FrameNavigation
-import models.AllVideos
+import models.AppState
 import ui.components.AutoSizeText
 
 /**
```
Comment (simonsasse) [inline GUI/src/main/kotlin/ui/screens/DiffScreen.kt:24]: adjust doc

File: GUI/src/main/kotlin/ui/screens/SelectVideoScreen.kt [hunk 4]
```diff
@@ -1,94 +1,68 @@
...
- *  TODO: remove this function
- *  This function is used to get the path of a file in the resources folder.
- *  @param name the name of the file
- *  @return [String] the path of the file
- */
-private fun getPath(name: String): String {
-    return FileSystems.getDefault().getPath("src", "main", "resources", name).toString()
-}
 
 @Composable
```
Comment (simonsasse) [inline GUI/src/main/kotlin/ui/screens/SelectVideoScreen.kt:12]: Since you're on it, add doc here as well :)

**Review by simonsasse — CHANGES_REQUESTED** [review]
Documentation-Police requests changes. 💅

**Comment by a-miscellaneous** [comment]
> Am I right, that the `state.value.copy(args)` args are only what changes in the state change, so that one does not have to write too much code when modifying the state? I yes, this approach seems good enough to me :)

yes that is correct

**Comment by a-miscellaneous** [comment]
lgtm

**Review by simonsasse — APPROVED** [review]

**Pull request merged**